### PR TITLE
refactor!(bloc_test): null safety

### PR DESCRIPTION
**Pull request by felangel**
## Status

**READY**

## Breaking Changes

YES

## Description

- refactor!(bloc_test): null safety (#2063) (unblocks #2044)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [X] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: examples/flutter_counter/pubspec.yaml [hunk 1]
```diff
@@ -19,12 +19,14 @@ dependency_overrides:
...
     path: ../../packages/flutter_bloc
+  bloc_test:
+    path: ../../packages/bloc_test
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
   bloc_test: ^7.1.0
-  mockito: ^4.0.0
+  mocktail: ">=0.0.1-dev.7 <0.1.0"
```
Comment (jorgecoca) [inline examples/flutter_counter/pubspec.yaml:29]: 👀

File: examples/flutter_counter/test/counter/view/counter_view_test.dart [hunk 2]
```diff
@@ -61,7 +61,7 @@ void main() {
       final decrementFinder = find.byKey(_decrementButtonKey);
       await tester.ensureVisible(decrementFinder);
       await tester.tap(decrementFinder);
-      verify(counterCubit.decrement()).called(1);
+      verify(counterCubit).called(#decrement).once();
```
Comment (jorgecoca) [inline examples/flutter_counter/test/counter/view/counter_view_test.dart:64]: Out of curiosity, but I want to make sure we cover all cases: if we use symbols, how can we verify a call with arguments?

For example, if the signature of decrement were `void decrement(int value)`, would you still do 👇 ?

```dart
verify(counterCubit).called(#decrement).once();
```

**Review by jorgecoca — APPROVED** [review]
LGTM, just had one question

File: examples/flutter_counter/test/counter/view/counter_view_test.dart (hunk 2 above)
Comment (felangel) [reply examples/flutter_counter/test/counter/view/counter_view_test.dart:64]: We can do
```dart
verify(counterCubit).called(#decrement).withArgs(positional: [any]).once();

```

**Pull request merged**